sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3540,6 +3541,32 @@ class Analyzer(override val catalogManager: CatalogManager)
     }
   }
 
+  /**
+   * Rule to mostly resolve, normalize and rewrite column names based on case sensitivity
+   * for alter table commands.
+   */
+  object ResolveAlterTableCommands extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
+      case a @ AlterTableDropColumns(r: ResolvedTable, colsToDrop) =>
+        val resolvedColsToDrop = colsToDrop.flatMap { col =>
+          resolveFieldNames(r.schema, col).orElse(Some(col))
+        }
+        a.copy(columnsToDrop = resolvedColsToDrop)
+    }
+
+    /**
+     * Returns the resolved field name if the field can be resolved, returns None if the column is
+     * not found. An error will be thrown in CheckAnalysis for columns that can't be resolved.
```

Comment:
Shall we fail here instead of in `CheckAnalysis`? Then we can avoid repeating the field name resolution logic in `CheckAnalysis`.

Reply:
Another idea is to follow `UnresolvedPartitionSpec`/`ResolvedPartitionSpec`. We add `UnresolvedFieldName` and `ResolvedFieldName`, to do the resolution in a general way like `ResolvePartitionSpec`

Reply:
I updated the PR with the `UnresolvedFieldName`/`ResolvedFieldName` approach. Please let me know what you think.

Reply:
Also, please let me know if you want to see more commands migrated in this PR.